samples/Common/cs/MediaPlayerFrameSource.cs

```diff
@@ -93,6 +101,16 @@ private void MediaPlayer_MediaOpened(MediaPlayer sender, object args)
                             (int)FrameWidth,
                             (int)FrameHeight);
 
+            // If a desired format was specified, create a staging VideoFrame to convert to when an image is obtained form the source before sending it out
+            if (m_desiredImageDescriptor != null)
```

Comment:
m_videoFrame already is a staging VideoFrame, maybe just use that instead.
```csharp
            if (m_desiredImageDescriptor != null)
            {
                m_videoFrame = new VideoFrame(
                    m_desiredImageDescriptor.SupportedBitmapPixelFormat,
                    m_desiredImageDescriptor.Width == -1 ? (int)FrameWidth : m_desiredImageDescriptor.Width,
                    m_desiredImageDescriptor.Height == -1 ? (int)FrameHeight : m_desiredImageDescriptor.Height,
                    m_desiredImageDescriptor.SupportedBitmapAlphaMode);
            }
            else
            {
                m_videoFrame = VideoFrame.CreateAsDirect3D11SurfaceBacked(
                        DirectXPixelFormat.B8G8R8A8UIntNormalized,
                        (int)FrameWidth,
                        (int)FrameHeight);
            }
```

Reply:
problem is that m_videoFrame needs to be a Direct3DSurface (backed VideoFrame), while if you specify a image format, I chose to only allow for SoftwareBitmap-backed VideoFrame for now

---
In reply to: [301351228](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/46#discussion_r301351228) [](ancestors = 301351228)

Reply:
(the need for Direct3DSurface comes from m_mediaPlayer.CopyFrameToVideoSurface(m_videoFrame.Direct3DSurface) )

---
In reply to: [301380121](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/46#discussion_r301380121) [](ancestors = 301380121,301351228)